invertedai_cpp/invertedai/drive_request.h

```diff
@@ -30,29 +30,104 @@ class DriveRequest {
   void update(const InitializeResponse &init_res);
+  /**
+   * Update the drive request with the information(agent_states,
+   * recurrent_states) in the drive response.
+   */
   void update(const DriveResponse &drive_res);
 
+  // getters
+  /**
+   * Get location string in IAI format.
+   */
   std::string location() const;
+  /**
+   * Get current states of all agents.
+   * x: [float], y: [float] corrdinate in meters;
+   * orientation: [float] in radians with 0 pointing along x
+   * and pi/2 pointing along y;
+   * speed: [float] in m/s.
+   */
   std::vector<AgentState> agent_states() const;
+  /**
+   * Get static attributes for all agents.
+   */
   std::vector<AgentAttributes> agent_attributes() const;
+  /**
+   * Get the states of traffic lights.
+   */
   std::vector<TrafficLightState> traffic_lights_states() const;
+  /**
+   * Get the recurrent states for all agents.
+   */
   std::vector<std::vector<double>> recurrent_states() const;
+  /**
+   * Check whether to return an image visualizing the simulation state.
+   */
   bool get_birdview() const;
+  /**
+   * Check whether to check predicted agent states for infractions.
+   */
   bool get_infractions() const;
+  /**
+   * Get random_seed.
+   */
   int random_seed() const;
 
+  // setters
+  /**
+   * Set location string in IAI format.
+   */
   void set_location(const std::string &location);
+  /**
+   * Set current states of all agents. The state must include x:
+   * [float], y: [float] corrdinate in meters orientation: [float] in radians
+   * with 0 pointing along x and pi/2 pointing along y and speed: [float] in
+   * m/s.
+   */
   void set_agent_states(const std::vector<AgentState> &agent_states);
+  /**
+   * Set static attributes for all agents.
+   */
   void
   set_agent_attributes(const std::vector<AgentAttributes> &agent_attributes);
+  /**
+   * Set the states of traffic lights. If the location contains traffic lights
+   * within the supported area, their current state should be provided here. Any
+   * traffic light for which no state is provided will be ignored by the agents.
+   */
   void set_traffic_lights_states(
       const std::vector<TrafficLightState> &traffic_lights_states);
+  /**
+   * Set the recurrent states for all agents, obtained from the
+   * previous call to drive() or initialize().
+   */
   void set_recurrent_states(
       const std::vector<std::vector<double>> &recurrent_states);
+  /**
+   * Set whether to return an image visualizing the simulation state.
+   * This is very slow and should only be used for debugging.
+   */
   void set_get_birdview(bool get_birdview);
+  /**
+   * Set whether to check predicted agent states for infractions.
+   * This introduces some overhead, but it should be relatively small.
+   */
   void set_get_infractions(bool get_infractions);
+  /**
+   * Set random_seed, which controls the stochastic aspects of agent behavior
+   * for reproducibility.
+   */
   void set_random_seed(int random_seed);
 };
 
```

Comment:
Are the getters and setters necessary for `drive-request`?
Especially, the `get_birdview` and `get_infraction` are very confusing.

Reply:
The same comment for `initialize_request.h` and  'location_info_request.h' since these are constructed by the user and don't seem to need getters (This is not critical though, we can talk about it later).

Reply:
Yes. I am also considering this, about the getters for the request class and the setters for the response class. They won't be used very frequently, but not harmful to keep.